bundles/org.eclipse.emfcloud.modelserver.common/src/org/eclipse/emfcloud/modelserver/common/patch/AbstractJsonPatchHelper.java

```diff
@@ -599,18 +603,17 @@ protected Object getValueFromList(final String jsonPath, final String[] segments
     */
    protected SettingValue getSettingFromCustomPath(final String modelURI, final String jsonPath)
       throws JsonPatchException {
-      int lastSegmentPos = jsonPath.lastIndexOf('/');
-      if (lastSegmentPos < 0 || lastSegmentPos >= jsonPath.length()) {
-         throw new JsonPatchException("Failed to parse Json Path: " + jsonPath);
-      }
-
       // The path may represent an EObject (without feature or index). In that case,
       // directly return it.
       EObject eObject = getEObject(modelURI, URI.createURI(jsonPath));
       if (eObject != null) {
          return new SettingValue(eObject);
       }
 
+      int lastSegmentPos = jsonPath.lastIndexOf('/');
+      if (lastSegmentPos < 0 || lastSegmentPos >= jsonPath.length()) {
```

Comment:
I know that this code is not changed but only moved, however as it was moved in this PR I think we should fix it.

The index of a character in a string cannot possibly be at the length of the string. I think probably this was intended:
```suggestion
      if (lastSegmentPos < 0 || lastSegmentPos >= (jsonPath.length() - 1)) {
```